src/wasm/kotlin/java/arcs/Addressable.kt

```diff
@@ -0,0 +1,30 @@
+package arcs
+
+/**
+ * Any object implementing this interface can be converted into a stable identifier. This
+ * could a platform specific pointer, identity hashcode, or id as long as it is guaranteed unique.
+ */
+interface Addressable
```

Comment:
I thought about it, but decided to restrict it to only Particle and Handle to avoid the map growing unavoidably from people adding addresses for everything (e.g. Strings, Numbers).  This is only for long-lived objects (alive until lifecycle ended). I'm open to other use cases, but wanted to err on the side of caution by making it explicit, instead of having an Any.toAddress()